packages/contracts-bedrock/test/DeployImplementations.t.sol

```diff
@@ -267,13 +266,142 @@ contract DeployImplementations_Test is Test {
+
+        address srDisputeGameFactoryImpl = address(0xc641A33cab81C559F2bd4b21EA34C290E2440C2B);
+        vm.etch(address(srDisputeGameFactoryImpl), hex"01");
+        deployImplementations.deployDisputeGameFactoryImpl(dii, dio);
+        assertEq(srDisputeGameFactoryImpl, address(dio.disputeGameFactoryImpl()));
+    }
+
+    function test_deployAtNonExistentRelease_reverts() public {
+        string memory unknownRelease = "op-contracts/v0.0.0";
+        dii.set(dii.release.selector, unknownRelease);
+
+        bytes memory expectedErr =
+            bytes(string.concat("DeployImplementations: failed to deploy release ", unknownRelease));
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deploySystemConfigImpl(dii, dio);
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployL1CrossDomainMessengerImpl(dii, dio);
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployL1ERC721BridgeImpl(dii, dio);
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployL1StandardBridgeImpl(dii, dio);
+
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployOptimismMintableERC20FactoryImpl(dii, dio);
+
+        // TODO: Uncomment the code below when OPContractsManager is deployed based on release. Superchain-registry
+        // doesn't contain OPContractsManager yet.
+        // dii.set(dii.superchainConfigProxy.selector, address(superchainConfigProxy));
+        // dii.set(dii.protocolVersionsProxy.selector, address(protocolVersionsProxy));
+        // vm.etch(address(superchainConfigProxy), hex"01");
+        // vm.etch(address(protocolVersionsProxy), hex"01");
+        // vm.expectRevert(expectedErr);
+        // deployImplementations.deployOPContractsManagerImpl(dii, dio);
+
+        dii.set(dii.proofMaturityDelaySeconds.selector, 1);
+        dii.set(dii.disputeGameFinalityDelaySeconds.selector, 2);
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployOptimismPortalImpl(dii, dio);
+
+        dii.set(dii.withdrawalDelaySeconds.selector, 1);
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployDelayedWETHImpl(dii, dio);
+
+        dii.set(dii.minProposalSizeBytes.selector, 1);
+        dii.set(dii.challengePeriodSeconds.selector, 2);
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployPreimageOracleSingleton(dii, dio);
+
+        address preImageOracleSingleton = makeAddr("preImageOracleSingleton");
+        vm.etch(address(preImageOracleSingleton), hex"01");
+        dio.set(dio.preimageOracleSingleton.selector, preImageOracleSingleton);
+        vm.expectRevert(expectedErr);
+        deployImplementations.deployMipsSingleton(dii, dio);
+
+        vm.expectRevert(expectedErr); // fault proof contracts don't exist at this release
+        deployImplementations.deployDisputeGameFactoryImpl(dii, dio);
+    }
+
+    function test_noContractExistsAtRelease_reverts() public {
```

Comment:
this test proves that you won't accidentally deploy implementations if the `op-contracts` tag exists but the contract doesn't.

Reply:
Great test cases 👌